### PR TITLE
Add Badge example

### LINKED ISSUE #144336: Add `Badge` usage example
### Steps to reproduce

In up-to-date Flutter, the following code should render a standard Material 3 badge indicator over the action icon:

```dart
    Scaffold(
        appBar: AppBar(
          title: const Text('Flutter Demo'),
          actions: [
            Badge.count(
              count: 3,
              child: IconButton(
                icon: const Icon(Icons.inbox),
                onPressed: () {},
              ),
            )
          ],
        ),
        body: const Center(
          child: Text(
            'Hello, Flutter!',
          ),
        ),
      )
```

### Expected results

Per https://m3.material.io/components/badges/overview, the badge should render "Anchor badges inside the icon bounding box, at the upper trailing edge of the icon". It should render correctly (and without additional configuration) at least in the following contexts:

"Badges are most commonly used within other components, such as navigation bar, navigation rail, top app bars, and tabs."

Here is a screenshot from material guidelines:

Screenshot 2024-02-28 at 11 33 51 AM

### Actual results

The badge is rendered too high and too far to the right of the icon.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Flutter Demo',
      theme: ThemeData(
        useMaterial3: true,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Flutter Demo'),
          actions: [
            Badge.count(
              count: 3,
              child: IconButton(
                icon: const Icon(Icons.inbox),
                onPressed: () {},
              ),
            )
          ],
        ),
        body: const Center(
          child: Text(
            'Hello, Flutter!',
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

Screenshot 2024-02-28 at 11 36 10 AM

</details>

### Logs

<details open><summary>Logs</summary>

```console
N/A
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
N/A
```

</details>

### PR DESCRIPTION
**Pull request by huycozy**
### Demo screenshot

Screenshot 2024-05-09 at 18 47 21

### Related issue
Fixes https://github.com/flutter/flutter/issues/144336

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/material/badge.dart [hunk 1]
```diff
@@ -24,6 +24,12 @@ import 'theme.dart';
 /// or a button's icon, as in [TextButton.icon]. The badge's default
 /// configuration is intended to work well with a default sized (24)
 /// [Icon].
+///
+/// {@tool dartpad}
+/// This example shows how to create a Badge and Badge.count that appears on an IconButton.
```
Comment (TahaTesser) [inline packages/flutter/lib/src/material/badge.dart:29]:
```suggestion
/// This example shows how to create a [Badge] with label and count
/// wrapped on an icon in an [IconButton].
```

File: examples/api/test/material/badge/badge.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/badge/badge.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Badge Test', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/material/badge/badge.0_test.dart:10]:
```suggestion
  testWidgets('Verify Badges have label and count', (WidgetTester tester) async {
```

**Review by TahaTesser — APPROVED** [review]
LGTM! With minor nits

File: packages/flutter/lib/src/material/badge.dart (hunk 1 above)
Comment (huycozy) [reply packages/flutter/lib/src/material/badge.dart]: thanks, it looks nicer

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks!

**Pull request merged**